python/ray/_raylet.pyx

```diff
@@ -460,8 +471,7 @@ cdef execute_task(
                             metadata_pairs, object_refs)
 
                     for arg in args:
-                        if isinstance(arg, RayError):
-                            raise arg
+                        raise_if_dependency_failed(arg)
```

Comment:
Do we know at this point what the current task's name is? It would be nice to print that or include it in the traceback, so that it is clear which task is actually failing (arg exceptions can get propagated through multiple functions).

Reply:
Task names are already included. For example, this is the chained tasks exception output;

```
ay.exceptions.RayTaskError: ray::dask:('timbre_algorithm-ee9fecbf0ee3e651065f20e250291f0a', 0, 0, 0) (pid=92366, ip=192.168.1.3)
  File "python/ray/_raylet.pyx", line 447, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 468, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 337, in ray._raylet.raise_if_dependency_failed
ray.exceptions.RayTaskError: ray::dask:('overlap-41117f6e20efebbabdc3ffb002c0d733', 0, 0) (pid=92368, ip=192.168.1.3)
  File "python/ray/_raylet.pyx", line 447, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 468, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 337, in ray._raylet.raise_if_dependency_failed
ray.exceptions.RayTaskError: ray::dask:('rechunk-split-65c67ebcb5e6fd3fa18779f44424653d', 0) (pid=92370, ip=192.168.1.3)
  File "python/ray/_raylet.pyx", line 447, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 468, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 337, in ray._raylet.raise_if_dependency_failed
ray.exceptions.RayTaskError: ray::dask:load_ten_min_file-06ddf54e-ccc1-4185-b37a-d59ccbb716ca (pid=92365, ip=192.168.1.3)
  File "python/ray/_raylet.pyx", line 519, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 520, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 1521, in ray._raylet.CoreWorker.store_task_outputs
  File "python/ray/_raylet.pyx", line 145, in ray._raylet.check_status
ray.exceptions.ObjectStoreFullError: Failed to put object a67dc375e60ddd1affffffffffffffffffffffff0100000001000000 in object store because it is full. Object size is 1382400257 bytes.
The local object store is full of objects that are still in scope and cannot be evicted. Tip: Use the `ray memory` command to list active objects in the cluster.
```

So, the bottom task failed, and that error is propagated to the upper parts of tasks (which is the same format as normal backtrace).

Do you think this is not sufficient? If so, can you tell me how you want this to look like? (I can modify code to achieve that)

Reply:
Ah I see, this looks good!